### ISSUE TITLE
Integration for Thunderbird 78

### ISSUE THREAD

**Issue by Stainzl** [kee-org/keebird#30]
Is support for Thunderbird 78 planned?

**Comment by dlech** [comment]
These are the changes that need to be made: https://developer.thunderbird.net/add-ons/updating/tb78

**Comment by kkapsner** [comment]
@dlech: I "borrowed" some code from keebird for [keepassxc-mail](https://github.com/kkapsner/keepassxc-mail) which runs in TB78. Either you can "borrow" it back or help me to get keepassxc-mail working with KeePassRPC.

But [apparently](https://github.com/kkapsner/keepassxc-mail/issues/11) you can get keepass to work with keepassxc-mail with the help of [keepassnatmsg](https://github.com/smorks/keepassnatmsg)...

**Comment by snodo** [comment]
My Thunderbird just auto-updated from 68 to 78
When will a compatible version be available?

**Comment by dlech** [comment]
> When will a compatible version be available?

Whenever someone has the time and motivation to do the work to make the required changes.

**Comment by Sardtok** [comment]
Would it make more sense to base a new add-on on the current Kee browser addon, which uses web extensions?

**Comment by dlech** [comment]
I think it would make more sense to look at https://github.com/kee-org/keebird/issues/30#issuecomment-684811116. Kee has diverged quite a bit from Keebird.

**Comment by luckyrat** [comment]
I agree that basing an updated Thunderbird extension on the latest Kee source code would most likely be overkill. There's not a huge amount of essential logic in there which needs to be in place for a successful MailExtension. However, pulling out just a few selected files (such as the ones that deal with the WebSocket connection and KeePassRPC messaging) and putting them into a new WebExtension should get most of the groundwork in place.

A wildcard option might be if someone could help out with implementing a native messaging connection for KeePassRPC - this is something that could then benefit the Kee browser extension too and thus reduce the number of plugins that need to be installed into KeePass. See https://github.com/kee-org/browser-addon/issues/23 for more details - obviously I'll be happy to help out anyone implementing that improvement to KeePassRPC but as I say in my latest comment there, it's the interaction between the browser (or Mail) extension, native messaging API, helper application and then KeePass/KeePassRPC that is most unknown to me so I think we would all be on a bit of a learning curve.

I've not followed Thunderbird development so perhaps anyone that knows the capabilities and practicalities of writing a MailExtension will tell me these are all crazy suggestions.

**Comment by kkapsner** [comment]
> However, pulling out just a few selected files (such as the ones that deal with the WebSocket connection and KeePassRPC messaging) and putting them into a new WebExtension should get most of the groundwork in place.

That was basically my starting point for keepassxc-mail. I pulled the scripts that communicate with KeePassXC from keepassxc-browser and then added the Thunderbird integration. But the later part is not as simple as you think. There is no API for "normal" MailExtensions to provide passwords (I opened https://bugzilla.mozilla.org/show_bug.cgi?id=1611700 for that reason) so I had to create an ["experiment"](https://github.com/kkapsner/keepassxc-mail/blob/master/experiment/implementation.js) that does that. This is the part where I took inspiration form keebird.
The "problem" with that is that you cannot submit a MailExtension with an experiment to addons.thunderbird.net (https://github.com/kkapsner/keepassxc-mail/issues/2) so it has to be side loaded.

If someone takes the RPC communication from Kee and the experiment from keepassxc-mail it might work. But to be honest I think it would be better to have one MailExtension which then can be used with KeePass and with KeePassXC. I have not looked into the details of RPC vs the Native Messaging approach of KeePassXC and do not know if one is superior.

**Comment by NicoPy** [comment]
So, what's the status of this issue ?
Is support for Thunderbird 78 planned ?

**Comment by kkapsner** [comment]
Auto update to Thunderbird 78 has started. I think you have to make a decision soon.

**Comment by simonmarwitz** [comment]
Meanwhile, you could disable thunderbird updates systemwide, as other extensions are affected as well:  https://support.mozilla.org/en-US/questions/1269626
or downgrade the profile after accidental updates:
https://support.cdn.mozilla.net/en-US/questions/1304550

**Comment by Q-Storm** [comment]
Are there any plans on integrate TB 78 or is there a roadmap on what is to do?
If not we have to think about an alternative extension.

**Comment by dlech** [comment]
These questions have already been answered:

Plans: https://github.com/kee-org/keebird/issues/30#issuecomment-692794285

What needs to be done: https://github.com/kee-org/keebird/issues/30#issuecomment-664452739

**Comment by dusscloud** [comment]
> If not we have to think about an alternative extension.

I didn't find any alternative yet, in order to have keepass working with Thunderbird 78. Is there any ?

**Comment by kkapsner** [comment]
> I didn't find any alternative yet, in order to have keepass working with Thunderbird 78. Is there any ?

https://github.com/kee-org/keebird/issues/30#issuecomment-684811116

**Comment by sparry0407** [comment]
Hi @dlech ,
Just wanted to say thanks for all the plug-ins development work you do: I personally rely completely on your work on a daily basis. I feel I should apologise for the tone of some of the comments here from other users; I wonder if any of these people realise how difficult it is to write any software, let alone security software and *for free*. Some politeness, gratitude and courtesy is surely a minimum requirement here. I am a developer and would like to help - I am totally new to this kind of development though.
Stephen

**Comment by wohauser** [comment]
Tried to test the pull request, but it does not install on 78 -- does not support TB 78
How to fix that ?
